lib/board.rb

```diff
@@ -21,20 +22,26 @@ def valid_coordinate?(coordinate)
   end
 
   def valid_placement?(ship, coordinates)
-    if coordinates.length != ship.length
+    if overlap?(coordinates) == true
+      false
+    elsif coordinates.length != ship.length
       false
     elsif consecutive_coordinates?(coordinates) == false
       false
-    elsif diagonal_coordinates?(coordinates)
+    elsif diagonal_coordinates?(coordinates) == true
       false
-    else
+    else 
       true
     end
   end
 
   def consecutive_coordinates?(coordinates)
-    letters = coordinates.map { |coordinate| coordinate[0].ord }
-    numbers = coordinates.map { |coordinate| coordinate[1..-1].to_i }
```

Comment:
I changed this for readability for myself while going over our code, and forgot to change it back